spec/09-lexical-structure.md

```diff
@@ -678,7 +678,7 @@ The variable substitution accepts the following syntax:
 * [*name*](#names)
 * [*variable-name*](#names)
 
-*expression* works the same way as in [variable name creation operator](10-expressions.md#variable-name-creation-operator).
+The `${ expression }` syntax is deprecated in Hack. The expression must be such that if the enclosing braces were removed and the `$` prepended, then the resulting expression would be a legal <i>`string-variable`</i>. For example, `"${y[123]}"` is legal because `$y[123]` is a legal <i>`string-variable`</i>. No whitespace or comments are allowed inside the braces.
```

Comment:
I agree. I'll delete the text for the not-supported-in-strict-mode feature entirely.

I am starting an internal document describing the differences between strict and lax modes; if we want to turn that into a spec in the future, but for now I'll just use it for my own purposes.